### PR TITLE
SQL backend

### PR DESCRIPTION
**Pull request by garrypolley**
- SQL backend is implemented and has unit tests
- Still need to add a test for the AuthenticationBackend
- Updated doc as needed

Note: mongoengine backend has not been implemented

### CODE REVIEW

File: django_generic_permissions/base_models.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        """Adds a permissoin for the user."""
+        raise NotImplementedError()
+
+    @classmethod
+    def remove_permission(cls, perm, user):
+        """Removes a permission from the user."""
+        raise NotImplementedError()
+
+    @classmethod
+    def has_permission(cls, perm, user):
```
Comment (garrypolley) [inline django_generic_permissions/base_models.py:18]: This should be implemented in the sql or removed.

Comment (garrypolley) [inline django_generic_permissions/base_models.py:18]: Will do in next pull request.

File: README.md [hunk 2]
```diff
@@ -25,13 +25,15 @@ def show_all_user_data(request):
```
Comment (garrypolley) [inline README.md:24]: Make this file a bit more readable.  will be done in next pull request.

File: django_generic_permissions/models.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# Need to verify that configuration is correct in settings.py
+if backend_db is not None and backend_db not in ACCEPTABLE_BACKENDS:
+    raise ImproperlyConfigured("{0} is not an accpetable backend only use one of thses: {1}"
+                                .format(backend_db, ACCEPTABLE_BACKENDS))
+
+
+# Define the location of the backends
+sql_backend = 'django_generic_permissions.sql_models'
+mogoengine_backend = 'django_generic_permissions.mongoengine_models'
```
Comment (hoot-dev) [inline django_generic_permissions/models.py:18]: Fix this, mogoengine no good.

File: django_generic_permissions/tests.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    def test_adding_multiple_permissions(self):
+        "UNIT make sure multiple permissions can be added correctly."
+        temp_user = FakeUser()
+        add_permission(self.to_add_perm_1, temp_user)
+        add_permission(self.to_add_perm_2, temp_user)
+        add_permission(self.to_add_perm_3, temp_user)
+
+        added_permissions = get_permissions(temp_user)
+
+        # Verify all permissions where added correctly and there are no extras
```
Comment (hoot-dev) [inline django_generic_permissions/tests.py:38]: Verify all permissions "were" added correctly...

**Comment by hoot-dev** [comment]
Aside from some minor spelling errors it looks good.

**Pull request merged**